### PR TITLE
Icecrown GunShipBattle, Fix for the cannons gain power on Cast an not On Hit Target

### PR DESCRIPTION
**Pull request by qwe900**
i did not found any about edit my old PullRequest so i tried now to add this again. 

## Changes Proposed:
Spelleffekt from Cast sets to Zero
Adding the Power Gain effect to spellscript in boss script  

## Issues Addressed:

- Closes 

## SOURCE:
https://youtu.be/nSDn2m0ThsI?t=33

## Tests Performed:
Tried on Windows 10 x64 with no Errors

## How to Test the Changes:
Go icecrown Citadel.
Go to Gunship Battle
EnterVehicle (cannon)
Press 1

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Spells/SpellMgr.cpp [hunk 1]
```diff
@@ -4733,6 +4733,13 @@ void SpellMgr::LoadDbcDataCorrections()
         spellInfo->EffectAmplitude[0] = 15000;
     });
 
+    //canon fix icc
+    ApplySpellFix({69399, 70172}, [](SpellEntry* spellInfo)
```
Comment (Kitzunu) [inline src/server/game/Spells/SpellMgr.cpp:4737]:
```suggestion
    // Cannon Blast
    ApplySpellFix({ 69399, 70172 }, [](SpellEntry* spellInfo)
```

**Review by Kitzunu — CHANGES_REQUESTED** [review]

**Comment by Kitzunu** [comment]
What's the reason for all the codestyle changes?

**Comment by qwe900** [comment]
> 
> 
> What's the reason for all the codestyle changes?
check-codestyle / check codestyle (ubuntu-20.04)  Failed...

**Review by qwe900 — COMMENTED** [review]
yes

**Comment by FrancescoBorzi** [comment]
@qwe900 the build is still failing, are you going to complete this PR?

**Comment by Nefertumm** [comment]
I really don't like the codestyle changes.
Also, the video you linked is not from WotLK, but I managed to find another one from Wrath era and the fix is correct.
https://www.youtube.com/watch?v=37XUADDw75U

But please, revert the codestyle changes. Thanks!

**Comment by acidmanifesto** [comment]
what exactly were the changes to the boss_icecrown_gunship_battle.cpp? White space changes are showing all around so the only thing i was able to see is the include directive #include "icecrown_citadel.h" being moved lower on quick glance.

**Comment by Nefertumm** [comment]
@acidmanifesto 
image
Tick that and you can see the real changes

**Comment by acidmanifesto** [comment]
> @acidmanifesto image Tick that and you can see the real changes

Thanks. im pretty boomer, so you taught be a new trick. i added in some review notes for him to stack them up as they were originally for readability.

**Comment by qwe900** [comment]
Switched now to github desktop, the other git software was doing some weird stuff idk.

**Comment by claudiodfc** [comment]
This PR is **WIP** as it is, this doesn't fix anything and messes up the codestyle of the whole CPP file.

**Comment by qwe900** [comment]
> This PR is **WIP** as it is, this doesn't fix anything and messes up the codestyle of the whole CPP file.

THis is bs, the codestyle was messed up by the github feature of edit file after just editet 2 lines an save it got this mess, i dont know how to revert this

**Comment by claudiodfc** [comment]
Honestly just delete branch and try again, it's pointless to revert to any commit since this solution isn't good.

**Comment by Winfidonarleyan** [comment]
Disable clang-format next time

**Pull request closed without merging**